src/libsyntax/parse/parser.rs

```diff
@@ -1198,24 +1198,29 @@ impl<'a> Parser<'a> {
                 };
 
                 let body = match p.token {
-                  token::Semi => {
-                    try!(p.bump());
-                    debug!("parse_trait_methods(): parsing required method");
-                    None
-                  }
-                  token::OpenDelim(token::Brace) => {
-                    debug!("parse_trait_methods(): parsing provided method");
-                    let (inner_attrs, body) =
-                        try!(p.parse_inner_attrs_and_block());
-                    attrs.extend(inner_attrs.iter().cloned());
-                    Some(body)
-                  }
+                    token::Semi => {
+                        try!(p.bump());
+                        if let token::DocComment(_) = p.token {
```

Comment:
While look ahead would be a viable approach if you were guaranteed to only get a single `DocComment` token, it doesn’t work here, because you effectively have to do infinite look-ahead now. This also is a very uncommonly used strategy in our parser: we just parse ahead and figure out what’s wrong when we hit a token we do not expect.